torch/_guards.py

```diff
@@ -146,11 +147,15 @@ def __hash__(self):
         return hash((self.name, self.source, id(self.create_fn)))
 
     def sort_key(self):
+        if isinstance(self.create_fn, functools.partial):
+            fn = self.create_fn.func
+        else:
+            fn = self.create_fn
```

Comment:
set `firstlineno` here.  The name `fn` is misleading.

Reply:
sure thing